maven/build-scan/publish/action.yml

```diff
@@ -0,0 +1,125 @@
+            # assuming directory name is ${{ env.BUILD_SCAN_DIR }}/<VERSION>/previous/<UUID>/scan.scan
+            currentVersion=$(basename ${scan%/previous/*})
+               
+            # check current version is not yet collected and matches a version pattern 
+            if [[ ! "${versions}" =~ .*"${currentVersion} ".* ]] && [[ "${currentVersion}" =~ ^[0-9]+(\.[0-9]+)*$ ]]; then
+              versions+="${currentVersion} "
+            else
+              echo "skipping ${currentVersion}"
+            fi
+          done        
+        fi
+        echo "versions=$versions" >> $GITHUB_OUTPUT
+      shell: bash
+    - name: Create Maven Project Structure
+      env:
+        PROJECT_DIR: 'maven-build-scan-publisher'
+      run: |
+        mkdir -p ${{ env.PROJECT_DIR }}/.mvn
+
+        cat > ${{ env.PROJECT_DIR }}/pom.xml << EOF
+        <project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+          <modelVersion>4.0.0</modelVersion>
+          <groupId>com.gradle</groupId>
+          <artifactId>${{ env.PROJECT_DIR }}</artifactId>
+          <version>1.0</version>
+          <name>Maven Build Scan Publisher</name>
+        </project>
+        EOF
+
+        cat > ${{ env.PROJECT_DIR }}/.mvn/extensions.xml.template << EOF
+        <?xml version="1.0" encoding="UTF-8"?>
+        <extensions>
+          <extension>
+            <groupId>com.gradle</groupId>
+            <artifactId>gradle-enterprise-maven-extension</artifactId>
+            <version>REPLACE_ME</version>
+          </extension>
+        </extensions>
+        EOF
+
+        cat > ${{ env.PROJECT_DIR }}/.mvn/gradle-enterprise.xml << EOF
+        <?xml version="1.0" encoding="UTF-8" standalone="yes" ?>
+        <gradleEnterprise 
+          xmlns="https://www.gradle.com/gradle-enterprise-maven" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="https://www.gradle.com/gradle-enterprise-maven https://www.gradle.com/schema/gradle-enterprise-maven.xsd">
+          <server>
+            <url>${{ inputs.gradle-enterprise-url }}</url>
+            <allowUntrusted>${{ inputs.gradle-enterprise-allow-untrusted }}</allowUntrusted>
+          </server>
+        </gradleEnterprise>
+        EOF
+      shell: bash
+    - name: Set up JDK 17
+      uses: actions/setup-java@v3
+      with:
+        java-version: '17'
+        distribution: 'temurin'
+    - name: Publish build scan
+      env:
+        BUILD_SCAN_DIR: '~/.m2/.gradle-enterprise/build-scan-data/'
+        GRADLE_ENTERPRISE_ACCESS_KEY: ${{ inputs.gradle-enterprise-access-key }}
+        PROJECT_DIR: 'maven-build-scan-publisher'
+      working-directory: ${{ env.PROJECT_DIR }}
+      run: |
+        set +e
+        
+        # iterate over extension versions
+        for version in ${{ steps.init.outputs.versions }}; do
```

Comment:
Is there a reason we can't include the logic from "collect extension versions" directly in this step?
In fact, could we just iterate over each scan file in turn, without separately collecting the versions?

Something like:
```
scans=$(find ${{ env.BUILD_SCAN_DIR }} -type f -name "scan.scan")
for scan in ${scans}; do        
  # Get the extension version for this scan
  version=$(basename ${scan%/previous/*})
  
  # set current version in project
  sed "s/REPLACE_ME/${version}/g" .mvn/extensions.xml.template > .mvn/extensions.xml
  
  # publish build scan
  mvn gradle-enterprise:build-scan-publish-previous
done
```

If that works, the logic would be much simpler!